src/vsr.zig

```diff
@@ -1098,3 +1102,251 @@ test "ViewChangeHeaders.view_for_op" {
+                // Follower: these headers are stored in the superblock's vsr_headers.
+                switch (chain) {
+                    .chain_sequence => {},
+                    // Gaps are due to either:
+                    // - entries before checkpoint, which are not repaired, or
+                    // - follower missed prepares and has not repaired headers. (Immediately after
+                    //   receiving a start_view this is not a concern, but the view_durable_update()
+                    //   may be delayed if another is in progress).
+                    .chain_view, .chain_gap => {
+                        assert(op <= options.op_checkpoint or !current.log_view_primary);
+                        break;
+                    },
+                    // Breaks are due to:
+                    // - entries before checkpoint, which are not repaired
+                    .chain_break => {
+                        assert(op <= options.op_checkpoint);
+                        break;
+                    },
+                }
+            } else if (suffix_done) {
+                // Add extra headers to the DVC. These are not required for correctness or
+                // availability, but including extra (correct) headers minimizes header repair at
+                // the new primary.
+                switch (chain) {
+                    .chain_sequence => {},
+                    .chain_view => {},
+                    // Outside of the log suffix, repair may not have been finished, so gaps and
+                    // breaks are possible. Non-same-view gaps may hide breaks.
+                    .chain_gap => break,
+                    .chain_break => break,
+                }
+            } else if (current.log_view_primary and command_durable == .start_view) {
+                switch (chain) {
+                    .chain_sequence => {},
+                    // Gaps to the right of the (durable) SV originate from:
+                    // 1. The primary (durable SV: 1,2,3) prepares several ops (4,5,6).
+                    // 2. However, the WAL writes are reordered such that some later ops (5,6)
+                    //    finish before an earlier op (4).
+                    // 3. Crash, recover. Start sending a DVC for the next view. Either:
+                    //    - There is a gap in the WAL at op=4, but this is to the right of the
+                    //      durable SV, so it may be safely skipped.
+                    //    - Same as above, except op=4 was a torn write (or bit rot).
+                    .chain_view, .chain_gap => assert(op + 1 > op_head_durable),
+                    // Breaks are impossible to the right of the durable SV — journal recovery uses
+                    // the durable SV to prune bad headers by their view numbers.
+                    .chain_break => unreachable,
+                }
+                suffix_done = op <= op_head_durable;
```

Comment:
When we use dvc_anchor for suffix, we cap it at `std.math.max(options.op_checkpoint, ...`, presumably because after checkpoint anything goes. Shouldn't be use the same capping for `op_head_durable` as well?

Reply:
And if we _didn't_ just recover, then installing the `start_view` would have truncated everything after the `start_view`.